datascience/util.py

```diff
@@ -53,3 +54,47 @@ def plot_cdf_area(rbound, lbound=None, mean=0, sd=1):
     plt.title("Normal Curve ~ ($\mu$ = {0}, $\sigma$ = {1}) "
               "{2} < z < {3}".format(mean, sd, llabel, rbound), fontsize=16)
     plt.show()
+
+
+def table_apply(table, func, subset=None):
+    """Applies a function to each column and returns a Table.
+
+    Uses pandas `apply` under the hood, then converts back to a Table
+
+    Parameters
+    ----------
+    table : instance of Table
+        The table to apply your function to
+    func : function
+        Any function that will work with DataFrame.apply
+    subset : list | None
+        A list of columns to apply the function to. If None, function
+        will be applied to all columns in table
+
+    Returns
+    -------
+    tab : instance of Table
+        A table with the given function applied. It will either be the
+        shape == shape(table), or shape (1, table.shape[1])
+    """
+    import pandas as pd
+    from datascience import Table
```

Comment:
Ah now I remember - I think I put the Table import in there because it was
raising import errors (presumably because of a circular import or
something). Nesting the Table import within the function makes it work
fine. But I put the pandas import at the top.

On Tue, Jan 12, 2016 at 9:27 PM, Chris Holdgraf <choldgraf@gmail.com> wrote:

> Sure thing - was just trying to minimize unnecessary imports but if you
> guys are cool with it I can pull 'em outside
>
> On Tue, Jan 12, 2016 at 8:56 PM, Sam Lau <notifications@github.com> wrote:
>
>> In datascience/util.py
>> <https://github.com/dsten/datascience/pull/175#discussion_r49551640>:
>>
>> > +    table : instance of Table
>> > +        The table to apply your function to
>> > +    func : function
>> > +        Any function that will work with DataFrame.apply
>> > +    subset : list | None
>> > +        A list of columns to apply the function to. If None, function
>> > +        will be applied to all columns in table
>> > +
>> > +    Returns
>> > +    -------
>> > +    tab : instance of Table
>> > +        A table with the given function applied. It will either be the
>> > +        shape == shape(table), or shape (1, table.shape[1])
>> > +    """
>> > +    import pandas as pd
>> > +    from datascience import Table
>>
>> Could we move these imports up to the top with the rest of the imports?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/dsten/datascience/pull/175/files#r49551640>.
>>
>
>